### PR TITLE
[4.0.] Remove obsolete and outdated trigger

### PR DESCRIPTION
**Pull request by roland-d**
### Summary of Changes
This removes the outdated and obsolete trigger. The correct trigger happens the line above.

### Testing Instructions
1. Go to Systems
2. Go to Plugins
3. Open plugin `Action Log - Joomla`
4. Click on `Save`
5. Notice you get the following error:
image
6. Apply the patch
7. Reload the page
8. Click on `Save`
9. Plugin is now saved

### Expected result
Plugin is saved

### Actual result
Class 'JEventDispatcher' not found error

### Documentation Changes Required
None

### CODE REVIEW

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on f2d08b1754eb4490f1a9056d9f2ee2fc02fb2087<br /><br />Note: The testing instructions only tell 1 example when that error happens. It happens also with other actions like e.g. enable/disable plugin with button in list view. This PR fixes all of them.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24941">issues.joomla.org/tracker/joomla-cms/24941</a>.</sub>

File: libraries/src/Table/Table.php [hunk 1]
```diff
@@ -1320,9 +1320,6 @@ public function checkIn($pk = null)
 		);
 		$this->getDispatcher()->dispatch('onTableAfterCheckin', $event);
 
-		$dispatcher = \JEventDispatcher::getInstance();
```
Comment (alikon) [inline libraries/src/Table/Table.php:1323]: i don't think it is correct, in this way you don't track anymore the `onAfterCheckin` event
should be coverted in j4 style
 ```php
Factory::getApplication()->triggerEvent('onAfterCheckin', array($this->_tbl));
```

**Review by alikon — CHANGES_REQUESTED** [review]

File: libraries/src/Table/Table.php (hunk 1 above)
Comment (richard67) [reply libraries/src/Table/Table.php:1323]: Yes, that works, too, I've just tested. Error also disappears when replacing lines 1323 and 1324 in the unpatched file by the line which @alikon suggests. So I think he is right.

**Comment by richard67** [comment]
@roland-d I think @alikon is right: Instead of removing lines 1323 and 1324, you should replace those 2 lines by the one line
`Factory::getApplication()->triggerEvent('onAfterCheckin', array($this->_tbl));`
like he suggests.

File: libraries/src/Table/Table.php (hunk 1 above)
Comment (alikon) [reply libraries/src/Table/Table.php:1323]: maybe the confusion arise cause there are 2 event : 

1. `onAfterCheckin `used by actionlog
2. `onTableAfterCheckin` unused by core

**Comment by richard67** [comment]
I have not tested this item.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24941">issues.joomla.org/tracker/joomla-cms/24941</a>.</sub>

**Comment by roland-d** [comment]
@alikon There is definitely a confusion with 2 triggers with basically the same name. What is the difference between them? One is J4 and the other is for J3?

@richard67 I have added the trigger as suggested

**Comment by richard67** [comment]
Would be confusing me if the was no confusion in J4 ;-) (joke)

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 485f9850673db0880d9188716308b1a9296684bf<br /><br />Note: The testing instructions only tell 1 example when that error happens. It happens also with other actions like e.g. enable/disable any plugin with the button in the plugins list view. This PR fixes all of them.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24941">issues.joomla.org/tracker/joomla-cms/24941</a>.</sub>

**Comment by alikon** [comment]
`onAfterCheckin` is worked by the  actionlog joomla plugin https://github.com/joomla/joomla-cms/blob/4.0-dev/plugins/actionlog/joomla/joomla.php#L924

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 485f9850673db0880d9188716308b1a9296684bf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24941">issues.joomla.org/tracker/joomla-cms/24941</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24941">issues.joomla.org/tracker/joomla-cms/24941</a>.</sub>

**Comment by infograf768** [comment]
FYI, there are other places where the error happens: for example when playing with custom admin menus. PR also solves issue there.

**Comment by infograf768** [comment]
Thanks.

**Comment by roland-d** [comment]
Thank you

**Pull request merged**